test/lib/next-modes/next-dev.ts

```diff
@@ -22,44 +20,47 @@ export class NextDevInstance extends NextInstance {
     if (this.childProcess) {
       throw new Error('next already started')
     }
-    // we don't use yarn next here as yarn detaches itself from the
-    // child process making it harder to kill all processes
-    const nextDir = path.dirname(resolveFrom(this.testDir, 'next/package.json'))
+    let startArgs = ['yarn', 'next']
 
-    this.childProcess = spawn('node', [path.join(nextDir, '/dist/bin/next')], {
-      cwd: this.testDir,
-      stdio: ['ignore', 'pipe', 'pipe'],
-      shell: false,
-      env: {
-        ...process.env,
-        NODE_ENV: '' as any,
-        __NEXT_TEST_MODE: '1',
-        __NEXT_RAND_PORT: '1',
-        __NEXT_TEST_WITH_DEVTOOL: '1',
-      },
-    })
+    if (this.startCommand) {
+      startArgs = this.startCommand.split(' ')
+    }
 
-    this.childProcess.stdout.on('data', (chunk) => {
-      const msg = chunk.toString()
-      process.stdout.write(chunk)
-      this._cliOutput += msg
-      this.emit('stdout', [msg])
-    })
-    this.childProcess.stderr.on('data', (chunk) => {
-      const msg = chunk.toString()
-      process.stderr.write(chunk)
-      this._cliOutput += msg
-      this.emit('stderr', [msg])
-    })
+    await new Promise<void>((resolve) => {
+      this.childProcess = spawn(startArgs[0], [...startArgs.slice(1)], {
```

Comment:
I dont think the `...` is necessary here. Maybe you're thinking of something like:

```js
const [cmd, ...args] = startArgs;
spawn(cmd, args)
```

Reply:
Right, updated in https://github.com/vercel/next.js/pull/33236/commits/0e8a91edaf8e58b1949059c0b1ed7392b5985b4f